### DISCUSSION TITLE
MVG will not accurate when if statement follow up with return statements

### DISCUSSION THREAD

**Discussion by JiahangLi** [sarnold/cccc#32] [Q&A]
It's a great open source Cyclomatic Complexity tool with extensive details about the program.
However, I found some issue while using it to test out our project. Here is a simple function that would help to illustrate the problem.

``` C
int main (){
    int a ,b ;
    a = 26; 
    b =10;
    if (a > 10 ||b < 45 && (a+b)< 55){
        return 1;
    } 
    return 2;
}
```

I might be wrong, but as far as I concern, this function should score MVG = 4, since we have three decision points while the measurement result by CCCC would be MVG = 5.

Thus, I did some experiment about it. 
1. MVG will go down to 4, if I remove the "return 2;" statement.
2. MVG will go down to 4, if the "return 1;" replaced by other code instead of a return statement.

It can lead to a big error rate, since some of big functions include several if statements and sprinkled with return statement as well.

**Comment by p2k-ko** [comment]
As I mentioned in issue #11 the C/C++ MVG metric is increased if a `return`-statement was found during lexing the function. 

**cccc.g**:
```diff
--- cccc.g
+++ cccc.g
@@ -202,7 +202,7 @@
 #token PROTECTED "protected" << ; >>
 #token PUBLIC "public" << ; >>
 #token REGISTER "register" << ; >>
-#token RETURN "return" << IncrementCount(tcMCCABES_VG); >>
+#token RETURN "return" << ; >>
 #token KW_SHORT "short" << ; >>
 #token SIGNED "signed" << ; >>
 #token SIZEOF "sizeof" << ; >>
```
**cccc_utl.cc**
``` diff
--- cccc_utl.cc	2017-03-03 07:20:06.129169794 +0100
+++ cccc_utl.cc	2017-03-03 07:21:51.938665925 +0100
@@ -299,6 +299,8 @@
     {
       lexical_counts_for_this_extent[i]=0;
     }
+  // Patch for starting the McCabe MVG metric with at least 1.
+  lexical_counts_for_this_extent[tcMCCABES_VG] = 1;
 
   if(allocate_lexcounts==true)
     {
```
